web/packages/teleterm/src/ui/AccessRequestCheckout/useAccessRequestCheckout.ts

```diff
@@ -337,7 +447,9 @@ export default function useAccessRequestCheckout() {
     isCollapsed,
     assumedRequests: getAssumedRequests(),
     toggleResource,
-    data: getPendingAccessRequestsPerResource(pendingAccessRequest),
+    data: getPendingAccessRequestsPerResource({
```

Comment:
In a similar vein to `getPendingAccessRequestsPerResource`, I think `data` needs to be rethought as well.

"Data" in itself is a super vague name. That'd be the most important thing, to find a name for it that better describes what `data` is.

When `data` is consumed in `web/packages/teleterm/src/ui/AccessRequestCheckout/AccessRequestCheckout.tsx`, it's usually filtered in different ways and the result is called `filteredData`, which is also pretty vague. Then we pass `filteredData` to some places and `data` to other. `data` is then filtered again.

As I understand, this is because sometimes we want to show items with kube clusters present and sometimes we want to ignore them and use only namespaces.

I think it'd be highly beneficial to find better names than `data` and `filteredData` which better describe in what scenarios certain values are used. Even better, once that is done, we can have separate types for them – this way we'll never pass `kube_cluster` items to places where they're not expected (think `Exclude<Data, {kind: 'kube_cluster'}>`.

Reply:
i think (hope) i resolved this with rename here: https://github.com/gravitational/teleport/pull/47722